File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.py
def solution(n):
    sam = []
    answer = 0
    
    while n > 0:
        sam.append(n % 3)
        n //= 3
        
    for i in range(len(sam)):
        answer += sam[-i - 1] * (3 ** i)
    
    return answer

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
import java.util.*;

class Solution {
    public int solution(int n) {
        int answer = 0;
        ArrayList<Integer> nList = new ArrayList<>();
        
        while (n > 0) {
            nList.add(n % 3);
            n /= 3;
        }
        
        int idx = nList.size() - 1;
        for (int i=0; i<=idx; i++) {
            answer += nList.get(idx - i) * Math.pow(3, i);
        }
        
        return answer;
    }
}